### ISSUE TITLE
pgbackrest command got stuck at  INFO: backup start archive during full backup

### ISSUE THREAD

**Issue by aherkarsatish11** [pgbackrest/pgbackrest#1253]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.20

2. PostgreSQL version: 10.x

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: Running crunchy operator 4.2.1 version - centos 7 base operating system,  1 master and 1 replica

4. Did you install pgBackRest from source or from a package?
Docker image : crunchy backrest ( crunchydata/pgo-backrest-repo:centos7-4.2.1 )
5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:

When I execute the full backup from leader node, it got stuck at archive and in debug log i see this query is running more than 
    70787 | 13:44:37.276087 | select datname, oid, datlastsysoid from pg_database" and in debug log i see below 👍 
2020-12-03 17:48:02.273 P00  DEBUG:     Db::executeSql(): bIgnoreError = <false>, bResult = <true>, strSql = select datname, oid, datlastsysoid from pg_database
2020-12-03 17:48:02.273 P00  DEBUG:     Db::connect(): bWarnOnError = <false>
2020-12-03 17:48:02.273 P00  DEBUG:     Db::connect=>: bResult = true
2020-12-03 17:48:02.273 P00  DEBUG:     postgres/client::pgClientQuery: (this: {host: {"/tmp"}, port: 5432, database: {"postgres"}, user: null, queryTimeout 1800000}, query: {"select datname, oid, datlastsysoid from pg_database"})
2020-12-03 17:48:02.373 P00  DEBUG:     postgres/client::pgClientQuery: => {VariantList}
2020-12-03 17:48:02.373 P00  DEBUG:     Db::executeSql=>: stryResult = ((postgres, 12888, 12887), (template1, 1, 12887), (template0, 12887, 12887), (ambrosia, 16386, 12887))
2020-12-03 17:48:02.373 P00  DEBUG:     Db->databaseMapGet=>: hDatabaseMap = [hash]
2020-12-03 17:48:02.374 P00  DEBUG:     Manifest->build(): bDelta = false, bOnline = true, bTablespace = [undef], hDatabaseMap = [hash], hTablespaceMap = [hash], iLevel = <0>, oLastManifest = [undef], oStorageDbMaster = [object], rhExclude = [undef], strFilter = [undef], strLevel = [undef], strParentPath = [undef], strPath = /pgdata/prod3-dw-sbak, strTimelineCurrent = 00000010, strTimelineLast = [undef]
2020-12-03 17:48:02.374 P00  DEBUG:     Storage::Storage->manifest(): strFilter = [undef], strPathExp = /pgdata/prod3-dw-sbak
2020-12-03 17:48:02.374 P00  DEBUG:     Storage::Storage->manifestJson(): strFilter = [undef], strPathExp = /pgdata/prod3-dw-sbak
2020-12-03 17:48:02.374 P00  DEBUG:     storage/storage::storageInfo: (this: {type: posix, path: {"/pgdata/prod3-dw-sbak"}, write: true}, fileExp: {"/pgdata/prod3-dw-sbak"}, param.ignoreMissing: true, param.followLink: false, param.noPathEnforce: false)
2020-12-03 17:48:02.374 P00  DEBUG:     storage/storage::storageInfo: => {StorageInfo}
2020-12-03 17:48:02.374 P00  DEBUG:     storage/storage::storageInfoList: (this: {type: posix, path: {"/pgdata/prod3-dw-sbak"}, write: true}, pathExp: {"/pgdata/prod3-dw-sbak"}, callback: (function *), callbackData: *void, param.errorOnMissing: true, param.sortOrder: 0, param.expression: null, param.recurse: false)

Can someone please help me, if I'm missing anything.

**Comment by dwsteele** [comment]
During the migration to C the interface between Perl and C made building a manifest for large numbers of files pretty slow.  You can check how many files you have to backup with:
```
find /pgdata/prod3-dw-sbak -type f | wc -l
```
If this number is more than a few hundred thousand then it may take quite a while.

2.20 was the last Perl version so we recommend upgrading to a more recent version, e.g. 2.30.

**Comment by aherkarsatish11** [comment]
Yes, there are more 2.7 million files in base directory.. we'll check with v2.30
